lib/cmdledit.g

```diff
@@ -751,10 +751,108 @@ GAPInfo.CommandLineEditFunctions.Functions.(INT_CHAR(' ')) :=
                 GAPInfo.CommandLineEditFunctions.Functions.SpaceDeletePrompt;
 BindKeysToGAPHandler(" ");
 
+
+BindGlobal("STANDARD_EXTENDERS", rec(
+  caseSensitive := function(cand, word)
+      local i, j, c;
+      i := Length(word);
+      while true do
+        if i = Length(cand[1]) then
+          break;
+        fi;
+        c := cand[1][i+1];
+        for j in [2..Length(cand)] do
+          if Length(cand[j]) > i and cand[j][i+1] = c then
+            j := j+1;
```

Comment:
This code confuses me. You increment `j`, but that's the loop variable, this has basically no effect, except perhaps for the last iteration, when `j=Length(cand)`, yes? So, while I think I see why you are doing this, that seems like an needlessly "clever" way to do this loop, which obscures what goes on.

Why not instead drop this, negate the if-condition, and the replace the `break;` by something like `commonPrefixFound := false`. So the loop would be roughly this (untested, just typed it into my browser):
```gap
  commonPrefixFound := true
  while commonPrefixFound and i < Length(cand[1])) do
    i := i + 1; # try finding a common prefix with one more character
    c := cand[1][i];
    for j in [2..Length(cand)] do
      if Length(cand[j]) < i or cand[j][i] <> c then
        commonPrefixFound := false;
        i := i - 1;
        break;
      fi;
    od;
  od;
```